### ISSUE TITLE
`solana-validator exit` feature request: force snapshot, delinquency ignore

### ISSUE THREAD

**Issue by nyetwurk** [solana-labs/solana#18863]
#### Problem

* When doing solana-validator exit, sometimes we have to wait a long time for a conjunction of snapshot request and leader slot gap.
* There is also no way to simply tell the validator to take snapshot "right now", e.g. for orderly shutdown, or if we want a snapshot to serve to someone else "right now"
* There is no way to force an `exit` when the delinquency is too high
* Non-voting validators also should not care if there is delinquency

#### Proposed Solutions

* `exit` should trigger a snapshot as soon as there is a gap in the leader schedule (or shortly before one)
*  There should be a way to manually trigger a snapshot (regardless of exit)
*  There should be way to ignore delinquency
*  Non-voting validators should always ignore delinquency

**Comment by ryoqun** [comment]
cc: @mvines

**Comment by nyetwurk** [comment]
Also, because there is no way to trigger a manual snapshot, `exit` can get stuck in a loop waiting for a leader gap, then waiting to take a snapshot, then waiting for *another* leader gap because the snapshot took too long to trigger.

**Comment by bji** [comment]
The first issue is largely addressed by incremental snapshots, which complete quickly and often enough that completing a snapshot shortly within a target window is the common case instead of the uncommon case.

The second issue I do not have experience with because I have never set min-idle-time "too low", not that I am aware of anyway.

The third issue is still an issue and would be a great feature enhancement.

The fourth issue I consider a non-issue, validators who are savvy enough to make correct decisions about whether or not to ignore cluster delinquency, can probably find fairly easy workarounds, and those validators not savvy enough to make correct decisions, are disallowed from making bad decisions.

The fifth issue I am not familiar with as I don't run an RPC server.

TL;DR: The largest problem, which is the difficulty of getting snapshots to complete in a timely enough manner within target windows, appears to largely be mitigated by incremental snapshots.

**Comment by michaelh-laine** [comment]
> The fourth issue I consider a non-issue, validators who are savvy enough to make correct decisions about whether or not to ignore cluster delinquency, can probably find fairly easy workarounds, and those validators not savvy enough to make correct decisions, are disallowed from making bad decisions.


The 4th issue was fixed in https://github.com/solana-labs/solana/pull/20367 over 6 months ago
